x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/IndexDeprecationChecks.java

```diff
@@ -30,14 +31,15 @@
  */
 public class IndexDeprecationChecks {
 
-    static DeprecationIssue oldIndicesCheck(IndexMetadata indexMetadata) {
+    static DeprecationIssue oldIndicesCheck(IndexMetadata indexMetadata, ClusterState clusterState) {
         // TODO: this check needs to be revised. It's trivially true right now.
         IndexVersion currentCompatibilityVersion = indexMetadata.getCompatibilityVersion();
-        if (currentCompatibilityVersion.before(IndexVersions.V_7_0_0)) {
+        // We intentionally exclude indices that are in data streams because they will be picked up by DataStreamDeprecationChecks
+        if (currentCompatibilityVersion.before(IndexVersions.V_8_0_0) && isNotDataStreamIndex(indexMetadata, clusterState)) {
```

Comment:
Do we need to worry about users potentially trying to upgrade their v7 data stream backing indices in 8.17 and not being able to because they are filtered out here? I'm pretty sure the answer is no, because they'll have to upgrade through 8.18 anyway to upgrade to 9.0.

Reply:
Yeah the deprecation info api and upgrade assistant are really only meaningful in the last release of a major version.